### PR TITLE
Forbid empty CountDown

### LINKED ISSUE #92196: `new CountDown(0)` is trappy and should be forbidden
Today in a few places we create a `new CountDown(0)`. This is kind of trappy because it will never return `true` from `#countDown()` or `#fastForward()` which risks never completing whatever action needs completing unless the caller handles the empty case specially.

We should refactor the places that do this so that they all avoid creating an empty `CountDown`, instead just handling the fact that the action is already complete:

- [ ] `TransportIndicesShardStoresAction$AsyncShardStoresInfoFetches`
- [ ] `ClearScrollController`
- [ ] `TransportBroadcastReplicationAction#doExecute`
- [ ] `Realms#usageStats`

Once we've done that, we can forbid the creation of an empty `CountDown`.

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today when using a `CountDown` callers must take responsibility for proper handling of the case where the count starts out at zero, because in this case the `countDown()` and `fastForward()` methods should never be called and will never return `true`. Fundamentally a `new CountDown(0)` doesn't really make sense, and it's a bit of a trap for the unwary, so with this commit we forbid it and adapt the handful of places which would create such an object.

Closes #92196

### CODE REVIEW

**Comment by DaveCTurner** [comment]
@elasticmachine test this please

**Comment by DaveCTurner** [comment]
Actually don't look at this too closely, I have a plan for fixing these bad callers in a better way. Watch this space...

**Comment by DaveCTurner** [comment]
This is good to review now. I recommend ignoring whitespace in the diffs.

File: server/src/main/java/org/elasticsearch/action/admin/indices/shards/TransportIndicesShardStoresAction.java [hunk 1]
```diff
@@ -279,5 +225,59 @@ public class Response {
                 }
             }
         }
+
+        void finish() {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/action/admin/indices/shards/TransportIndicesShardStoresAction.java:229]: This method had no business being a member of `InternalAsyncFetch`, this change just moves it up a level.

File: server/src/main/java/org/elasticsearch/action/admin/indices/shards/TransportIndicesShardStoresAction.java [hunk 2]
```diff
@@ -279,5 +225,59 @@ public class Response {
...
+                v.replaceAll((s, l) -> {
+                    CollectionUtil.timSort(l);
+                    return List.copyOf(l);
+                });
+                return Map.copyOf(v);
+            });
+            listener.onResponse(new IndicesShardStoresResponse(Map.copyOf(indicesStatuses), List.copyOf(failures)));
+        }
+
+        private AllocationStatus getAllocationStatus(String index, int shardID, DiscoveryNode node) {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/action/admin/indices/shards/TransportIndicesShardStoresAction.java:260]: This method had no business being a member of `InternalAsyncFetch`, this change just moves it up a level.

File: server/src/main/java/org/elasticsearch/action/admin/indices/shards/TransportIndicesShardStoresAction.java [hunk 3]
```diff
@@ -279,5 +225,59 @@ public class Response {
...
+                    }
+                }
+            }
+            return AllocationStatus.UNUSED;
+        }
+
+        /**
+         * A shard exists/existed in a node only if shard state file exists in the node
+         */
+        private static boolean shardExistsInNode(final NodeGatewayStartedShards response) {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/action/admin/indices/shards/TransportIndicesShardStoresAction.java:279]: This method had no business being a member of `InternalAsyncFetch`, this change just moves it up a level.

**Review by thecoop — APPROVED** [review]
LGTM, good changes here

**Pull request merged**